CMU-Robotics-Club/RoboBuggy
# Use Gradle to build Alice

Some say that Eclipse-as-a-build-system may not scale. These detractors may be wrong, but it certainly doesn't allow unit tests, perform continuous integration, or verify your pull request won't break the build. 

This merge request will track main, but rather than eclipse it will use gradle to build Alice and BuggyRos.

**This is ready for Beta testing**

Please see surface_src\java_src\README.md for more information.

Steps to test
==========
    
    $> cd MyBuggyRepo
    $> git fetch # Download the stuff, but doesn't update HEAD pointer.
    $> git stash 
    $> git checkout msebek_gradle_build
    $> cd real_time/surface_src/java_src/
    $> ./gradlew build
    $> cd Alice
    $> ../gradlew run
    <Expected behavior: GUI pops up.>
    <Unexpected behavior: Everything else>

    $> cd .. # Back to java_src
    $> gradlew eclipse
    $> (import the generated projects into eclipse)

Once the gui-appearing has been validated, try to validate sending/receiving serial data. 
If this does not work, add the line 'run files('library/RXTXcomm.jar' at Alice\build.gradle:17, to (maybe) add a runtime dependency on Rxtx, which might be a good thing. 

Required Reviewers: @VasuAgrawal @VivaanBahl @TrevorDecker 
Optional Reviewers: @ihartwig @ayeohmy
